src/contexts/app.context.js

```diff
@@ -3,14 +3,28 @@ import React, { createContext, useContext, useState, useEffect } from 'react'
 import PropTypes from 'prop-types'
 import constants from '../util/constants'
 import { useHistory } from 'react-router-dom'
+import { themeDark, themeLight } from '../util/themes'
 
 const PagesContext = createContext()
 
 export function PagesProvider({ children }) {
   const history = useHistory()
+  const [theme, setTheme] = useState(
+    JSON.parse(localStorage.getItem('@hoco/theme'))
+      ? JSON.parse(localStorage.getItem('@hoco/theme'))
+      : themeLight
+  )
+
+  const handleTheme = (themeName) => {
+    const currentTheme = themeName === 'light' ? themeLight : themeDark
+    localStorage.setItem('@hoco/theme', JSON.stringify(currentTheme))
+
+    setTheme(JSON.parse(localStorage.getItem('@hoco/theme')))
+  }
```

Comment:
Acredito que esse código pode ser simplificado.

```suggestion
  const getTheme = () => useCallback({
    return JSON.parse(localStorage.getItem('@hoco/theme'))
  }, [])
  
  const [theme, setTheme] = useState(currentTheme ? currentTheme : themeLight)

  const handleTheme = (themeName) => {
    localStorage.setItem('@hoco/theme', JSON.stringify(THEMES[themeName]))

    setTheme(getTheme())
  }
```
A constante themes seria definida em outro arquivo da seguinte forma (`util/themes.js`:

```js
const light = {
  ...
};

const dark = {
  ...
};

export const THEMES = {
  light,
  dark
};
```

export default THEMES;